src/utilities/scripts/aws_10x.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+
+import argparse
+import warnings
+import re
+
+from utilities.alignment.run_10x_count import reference_genomes, deprecated
+import utilities.s3_util as s3u
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Create a shell to run alignment jobs"
+        " with 10x for multiple samples all together"
+    )
+    # required arguments
+    requiredNamed = parser.add_argument_group("required arguments")
+
+    requiredNamed.add_argument(
+        "--taxon",
+        choices=list(reference_genomes.keys()),
+        required=True,
+        help="Reference genome for the alignment run, "
+        "selected from the reference_genomes dictionary keys from "
+        "alignment.run_10x_count.py",
+    )
+
+    requiredNamed.add_argument(
+        "--s3_input_path",
+        required=True,
+        help="The folder containing sample folders, "
+        "each of which have fastq.gz files to align",
+    )
+
+    requiredNamed.add_argument(
+        "--s3_output_path",
+        required=True,
+        help="The folder to store the alignment results",
+    )
+
+    # optional arguments
+    parser.add_argument(
+        "--branch", default="master", help="Branch of utilities repo to use"
+    )
+
+    parser.add_argument(
+        "script_args",
+        nargs=argparse.REMAINDER,
+        help="Extra arguments are passed to run_10x_count",
+    )
+
+    args = parser.parse_args()
+
+    # check if the input genome is valid
+    if args.taxon in reference_genomes:
+        if args.taxon in deprecated:
+            warnings.warn(
+                f"The name '{args.taxon}' will be removed in the future,"
+                f" start using '{deprecated[args.taxon]}'"
+            )
+    else:
+        raise ValueError(f"unknown taxon {args.taxon}")
+
+    # get the list of sample folder paths under the input folder
+    s3_input_bucket, s3_input_prefix = s3u.s3_bucket_and_key(args.s3_input_path)
+    s3_input_prefix += "/"
+    sample_folder_pattern = re.compile(
+        s3_input_prefix + "([a-zA-Z0-9]+_)+10X_\d+_\d+.*/$"
```

Comment:
This regex has a few issues, but it's moot because I don't think we can use a regex here at all. There's very little constraining what these names will look like. It's basically a free text entry from the person running the experiment.

I think you have to just assume every folder under the given path is a sample. Or if you want to be extra careful, check each one to make sure it contains fastq files. It will be up to the user to organize their data for this.

Reply:
Then I'll just delete the pattern. When I just got started I tried to run non-fastq files with the alignment codes and it just outputs the inputs without any changes or error, so even if there're files other than fastq in the folder probably it shouldn't be disturb too much.